rig-core/src/providers/perplexity.rs

```diff
@@ -0,0 +1,248 @@
+//! Perplexity API client and Rig integration
+//!
+//! # Example
+//! ```
+//! use rig::providers::perplexity;
+//!
+//! let client = perplexity::Client::new("YOUR_API_KEY");
+//!
+//! let llama_3_1_sonar_small_online = client.completion_model(perplexity::LLAMA_3_1_SONAR_SMALL_ONLINE);
+//! ```
+
+use crate::{
+    agent::AgentBuilder,
+    completion::{self, CompletionError},
+    extractor::ExtractorBuilder,
+    json_utils,
+    model::ModelBuilder,
+    rag::RagAgentBuilder,
+    vector_store::{NoIndex, VectorStoreIndex},
+};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+// ================================================================
+// Main Cohere Client
+// ================================================================
+const PERPLEXITY_API_BASE_URL: &str = "https://api.perplexity.ai/chat/completions";
+
+#[derive(Clone)]
+pub struct Client {
+    base_url: String,
+    http_client: reqwest::Client,
+}
+
+impl Client {
```

Comment:
Would be nice to add a `from_env` helper method similar to the one for the OpenAI client